src/ORM/Association/BelongsToMany.php

```diff
@@ -717,20 +717,21 @@ function () use ($sourceEntity, $targetEntities, $options) {
      * this association
      * @param bool $cleanProperty whether or not to remove all the objects in $targetEntities
      * that are stored in $sourceEntity
+     * @param array $options list of options to be passed to the internal `delete` call
      * @throws \InvalidArgumentException if non persisted entities are passed or if
      * any of them is lacking a primary key value
      * @return void
      */
-    public function unlink(EntityInterface $sourceEntity, array $targetEntities, $cleanProperty = true)
+    public function unlink(EntityInterface $sourceEntity, array $targetEntities, $cleanProperty = true, array $options = [])
```

Comment:
@ndm2 Change `$cleanProperty` itself to `$options`. Though for BC we would have to allow `$options` to be passed a boolean and hence can't typehint `$options` as array.